File: day_01/part1.py
##############################################################################
# 2018 Advent of Code
# Day: 1
# Part: 1
# Answer: 553
#
# @author Matt Adorjan
##############################################################################

# Read frequency changes in to a variable
with open('./input.txt', 'r') as f:
    data_in = f.readlines()

# Set initial frequecy as 0
FREQ = 0

# For each of the frequency changes, add to the current frequency value
for delta in data_in:
    FREQ = FREQ + int(delta.strip())    # Convert from str -> int and strip off line endings (\n)

print("Final frequency after calibration is: {}".format(FREQ))

File: day_01/go/part1.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
	"strconv"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	dat, err := ioutil.ReadFile("../input.txt")
	check(err)
	lines := strings.Split(string(dat), "\n")

	var FREQ int64
	FREQ = 0

	for _, line := range lines {
		line, err := strconv.ParseInt(line, 10, 64)
		check(err)
		FREQ = FREQ + line
	}
	fmt.Println("Final frequency after calibration is:", FREQ)
}